### DISCUSSION TITLE
react-instantsearch-router-nextjs dependency is not compatible with NextJS 14

### DISCUSSION THREAD

**Discussion by Isaac-Tait** [algolia/instantsearch#5933] [Q&A] [answered]
When attempting to install the [react-instantsearch-router-nextjs](https://www.npmjs.com/package/react-instantsearch-router-nextjs) dependency I get the following error:

```
code ERESOLVE
npm ERR! ERESOLVE unable to resolve dependency tree
npm ERR! 
npm ERR! While resolving: page_router_royal_ridges@0.1.0
npm ERR! Found: next@14.0.2
npm ERR! node_modules/next
npm ERR!   next@"14.0.2" from the root project
npm ERR! 
npm ERR! Could not resolve dependency:
npm ERR! peer next@">= 9 && < 14" from react-instantsearch-router-nextjs@7.3.0
npm ERR! node_modules/react-instantsearch-router-nextjs
npm ERR!   react-instantsearch-router-nextjs@"*" from the root project
```

It appears that [this dependency is necessary](https://www.algolia.com/doc/guides/building-search-ui/going-further/server-side-rendering/react/#with-nextjs) to implement Algolia, so what can I do to work around this issue?

**Comment by Haroenv** [comment]
Ah sorry, we haven't yet updated the peer dependency, but it does work. For now you can ignore the warning, and we'll make a new version that does mark itself as compatible.

**Comment by Isaac-Tait** [comment]
I ran `--force` when installing but it is failing the build in Netlify. Here is the build error message:

```
11:36:43 AM: npm ERR! code ERESOLVE
11:36:43 AM: npm ERR! ERESOLVE could not resolve
11:36:43 AM: npm ERR!
11:36:43 AM: npm ERR! While resolving: react-instantsearch-router-nextjs@7.3.0
11:36:43 AM: npm ERR! Found: next@14.0.2
11:36:43 AM: npm ERR! node_modules/next
11:36:43 AM: npm ERR!   next@"14.0.2" from the root project
11:36:43 AM: npm ERR!
11:36:43 AM: npm ERR! Could not resolve dependency:
11:36:43 AM: npm ERR! peer next@">= 9 && < 14" from react-instantsearch-router-nextjs@7.3.0
11:36:43 AM: npm ERR! node_modules/react-instantsearch-router-nextjs
11:36:43 AM: npm ERR!   react-instantsearch-router-nextjs@"^7.3.0" from the root project
11:36:43 AM: npm ERR!
11:36:43 AM: npm ERR! Conflicting peer dependency: next@13.5.6
11:36:43 AM: npm ERR! node_modules/next
11:36:43 AM: npm ERR!   peer next@">= 9 && < 14" from react-instantsearch-router-nextjs@7.3.0
11:36:43 AM: npm ERR!   node_modules/react-instantsearch-router-nextjs
11:36:43 AM: npm ERR!     react-instantsearch-router-nextjs@"^7.3.0" from the root project
11:36:43 AM: npm ERR!
```

**Reply by Haroenv** [reply]
the version that marks compatibility will be released tomorrow :)

**Reply by Isaac-Tait** [reply]
Outstanding!

**Reply by Haroenv** [reply]
sorry, I forgot it's Black Friday week, release after the code freeze, but you should be able to run --force on netlify too

**Reply by Isaac-Tait** [reply]
Taking a look [at the Netlify docs](https://docs.netlify.com/configure-builds/troubleshooting-tips/#build-fails-on-peer-dependency-conflict) I tried this 
![CleanShot 2023-11-21 at 10 22 08](https://github.com/algolia/instantsearch/assets/14932877/b202fe17-fd79-4452-ba31-80bd1bdb7b71)
but am still getting the npm build error. Did I do the 'NPM_FLAG' right? 

EDIT: Also tried `NPM_FLAG_BUILD` after [reading this](https://docs.npmjs.com/cli/v8/using-npm/config#command-line-flags)

**Reply by Haroenv** [reply]
I think it's `NPM_FLAGS`, not `NPM_FLAG`

**Comment by Haroenv** [comment, marked as answer]
Latest version which has its compatibility marked is released now
